File: 2012/submissions/Pat Smuk/1/C/Problem1.c
/*
 * A plain C solution to problem 1.
 * By Pat Smuk.
 * Tested on Linux Mint with GCC 4.7.2.
 */

#include <stdio.h>
#include <string.h>

#define MAX_DATA 500    // 500 bytes is enough for anyone!


void removeLF(char *str)
{
    int i = 0;
    while (str[i] != '\0') ++i; // find the terminator
    if (i == 0) return;         // prevent weirdness if passed NULL
    str[i-1] = 0;               // set previous char to 0
}

int main(int argc, char *argv[])
{
    char word [MAX_DATA];
    printf("Enter the word to search for: ");
    
    if (fgets(word, MAX_DATA-1, stdin) == NULL) {
        puts("Could not read word.");
        return 1;
    }
    
    removeLF(word);
    
    char text [MAX_DATA];
    printf("Enter the text to search: ");
    
    if (fgets(text, MAX_DATA-1, stdin) == NULL) {
        puts("Could not read text.");
        return 1;
    }
    
    removeLF(text);
    
    if (strstr(text, word) == NULL) {
        printf("The word \"%s\" was not found in the text.\n", word, text);
        return 1;
    }
    
    printf("First index of \"%s\" in text: %ld\n", word, (long)(strstr(text, word) - text));
    
    return 0;
}

File: 2012/submissions/Pat Smuk/1/Objective-C/Problem1.m
/*
 *	Solution to problem 1 in Objective-C.
 *	By Pat Smuk.
 */

#import <Foundation/Foundation.h>
#import <stdio.h>

#define MAX_DATA 500


void removeLF(char *str)
{
	int i = 0;
	while (str[i] != '\0') ++i;
	str[i-1] = '\0';
}

int main(int agc, char *argv[])
{
	NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
	
	printf("Enter a word to find: ");
	
	char search_buf [MAX_DATA];
	if (fgets(search_buf, MAX_DATA-1, stdin) == NULL) {
		puts("Could not read data.");
		return 1;
	}
	
	NSString *search = [NSString stringWithCString:search_buf encoding:NSUTF8StringEncoding];
	search = [search substringToIndex:search.length-1];
	
	printf("Enter the text to search: ");
	
	char text_buf [MAX_DATA];
	if (fgets(text_buf, MAX_DATA-1, stdin) == NULL) {
		puts("Could not read data.");
		return 1;
	}
	
	NSString *text = [NSString stringWithCString:text_buf encoding:NSUTF8StringEncoding];
	text = [text substringToIndex:text.length-1];
	
	NSUInteger index = [text rangeOfString:search].location;
	NSString *result = [NSString stringWithFormat:@"Word \"%@\" found at index: %u", search, index];
	puts([result UTF8String]);
	
	[pool release];
	return 0;
}